minpy/array_variants/numpy/numpy_core.py

```diff
@@ -116,25 +178,29 @@ def def_grads(reg, prims):
 
     # Reducing functions.
     prims('sum').def_grad(_sum_grad)
-    #prims('max').def_grad(_chooser_grad)
-    #prims('min').def_grad(_chooser_grad)
-    #prims('amax').def_grad(_chooser_grad)
-    #prims('amin').def_grad(_chooser_grad)
-    prims('max').def_grad_zero()
-    prims('min').def_grad_zero()
-    prims('amax').def_grad_zero()
-    prims('amin').def_grad_zero()
-
+    prims('max').def_grad(
+        lambda ans, a, axis=None, out=None, keepdims=False: _minpy_amax_amin_grad(ans, a, axis, out, keepdims)
+    )
+    prims('amax').def_grad(
+        lambda ans, a, axis=None, out=None, keepdims=False: _minpy_amax_amin_grad(ans, a, axis, out, keepdims)
+    )
+    prims('min').def_grad(
+        lambda ans, a, axis=None, out=None, keepdims=False: _minpy_amax_amin_grad(ans, a, axis, out, keepdims)
+    )
+    prims('amin').def_grad(
+        lambda ans, a, axis=None, out=None, keepdims=False: _minpy_amax_amin_grad(ans, a, axis, out, keepdims)
+    )
+    
```

Comment:
There is an extra wrap of lambda here. You could simply define a function, say `_minpy_amax_amin_grad` which has the same signature of `lambda ans, a, axis, out, keepdims:`. Also `_chooser_grad` sounds better (from autograd package). It is not only useful for amax, amin, but also useful for other "choose from array content" function.